CSharp/Library/Microsoft.Bot.Connector.Shared/StateAPI/Models/BotData.cs

```diff
@@ -21,9 +21,8 @@ public partial class BotData
         /// <summary>
         /// Initializes a new instance of the BotData class.
         /// </summary>
-        public BotData(string eTag = default(string), object data = default(object))
+        public BotData(string eTag = @"*", object data = default(object))
```

Comment:
I don't believe any updates are needed to our docs.  The only time ETag=* is mentioned in our docs is to say that ETag=* in GET response (i.e., GetUserData, GetConversationData, or GetPrivateConversationData) indicates that no state data exists in the store for the user and/or conversation that is specified in a "Get..." method call. The docs do not mention anything about setting ETag=* in a SET request.